enderio-base/src/main/java/com/enderio/base/common/block/glass/GlassBlocks.java

```diff
@@ -54,10 +55,39 @@ public GlassBlocks(Registrate registrate, String name, String english, GlassColl
         BLACK = register(registrate, name.concat("_black"), "Black ".concat(english), DyeColor.BLACK);
     }
 
-    private static ResourceLocation getModelFile(String name) {
-        return name.contains("clear_glass") ? EnderIO.loc("block/clear_glass") : EnderIO.loc("block/fused_quartz");
+    private ResourceLocation getModelFile() {
+        return explosionResistant ? EnderIO.loc("block/fused_quartz") : EnderIO.loc("block/clear_glass");
     }
 
+    private static String createGlassName(GlassIdentifier identifier) {
+        StringBuilder main = new StringBuilder();
+        if (identifier.explosion_resistance()) {
+            main.append("fused_quartz");
+        } else {
+            main.append("clear_glass");
+        }
+        StringBuilder modifier = new StringBuilder();
+        modifier.append(identifier.lighting().shortName());
+        modifier.append(identifier.collisionPredicate().shortName());
+        if (modifier.length() != 0) {
+            main.append("_");
+            main.append(modifier);
+        }
+        return main.toString();
+    }
+    private static String createEnglishGlassName(GlassIdentifier identifier) {
+        StringBuilder main = new StringBuilder();
+        if (identifier.lighting() != GlassLighting.NONE) {
+            main.append(identifier.lighting().englishName());
+            main.append(" ");
```

Comment:
Could just add the space at the start of Fused/Clear glass? Might read a little nicer?

Reply:
If we would to that, a space might be in front if GlassLighting is none. Then I would have to trim it, which I don't really want.
I thinks it's readable enough

Reply:
ahh true true, nevermind 😅